File: check_matches.py

import os
import json
import firebase_admin
from firebase_admin import credentials, firestore

def check_recent_matches():
    if not firebase_admin._apps:
        cred_path = "backend/service-account-new.json"
        if os.path.exists(cred_path):
            cred = credentials.Certificate(cred_path)
            firebase_admin.initialize_app(cred)
        else:
            print("Credentials not found.")
            return

    db = firestore.client()
    matches = db.collection('match_stats').order_by('date', direction=firestore.Query.DESCENDING).limit(5).stream()
    
    print("Recent matches in match_stats:")
    for m in matches:
        data = m.to_dict()
        print(f"ID: {m.id} | Date: {data.get('date')} | Teams: {data.get('home_team')} x {data.get('away_team')} | Status: {data.get('status')}")

if __name__ == "__main__":
    check_recent_matches()

File: scratch/check_matches.js
const admin = require('firebase-admin');
const serviceAccount = require('./backend/service-account.json');

if (!admin.apps.length) {
    admin.initializeApp({
        credential: admin.credential.cert(serviceAccount)
    });
}

const db = admin.firestore();

async function checkMatches() {
    console.log("Checking upcoming matches for Botafogo...");
    const threshold = new Date();
    threshold.setHours(threshold.getHours() - 3);

    const snapshot = await db.collection('matches')
        .where('date', '>=', threshold.toISOString())
        .orderBy('date', 'asc')
        .limit(100)
        .get();

    if (snapshot.empty) {
        console.log("No upcoming matches found in DB.");
        return;
    }

    const botafogoMatches = [];
    snapshot.forEach(doc => {
        const data = doc.data();
        if (data.home_team?.includes('Botafogo') || data.away_team?.includes('Botafogo')) {
            botafogoMatches.push({
                id: doc.id,
                championship: data.championship,
                home: data.home_team,
                away: data.away_team,
                date: data.date
            });
        }
    });

    console.log(`Found ${botafogoMatches.length} upcoming matches for Botafogo:`);
    botafogoMatches.forEach(m => {
        console.log(`- [${m.championship}] ${m.home} vs ${m.away} on ${m.date}`);
    });
}

checkMatches().catch(console.error);
